python/ray/workflow/execution.py

```diff
@@ -117,8 +110,19 @@ def get_output(workflow_id: str, name: Optional[str]) -> ray.ObjectRef:
             "actor. The workflow could have already failed. You can use "
             "workflow.resume() to resume the workflow."
         ) from e
-    output = ray.get(workflow_manager.get_output.remote(workflow_id, name))
-    return flatten_workflow_output(workflow_id, output)
+
+    try:
+        wf_store = workflow_storage.WorkflowStorage(workflow_id)
+        tid = wf_store.inspect_output(name)
+        if tid is not None:
+            return load_step_output_from_storage.remote(workflow_id, name)
+    except Exception:
+        # If the workflow just starts running, then it could be normal that the workflow
+        # cannot be found in the storage.
+        if not ray.get(workflow_manager.is_workflow_running.remote(workflow_id)):
+            raise
```

Comment:
I'm thinking about one corner case here. What if:

1. workflow just started
2. it goes to Exception
3. workflow has been finished
4. is_workflow_running return false
5. raise

Should we call workflow_manager.get_output and move the checking and other things there?